graphql-dgs/src/main/java/com/netflix/graphql/dgs/Trivial.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import graphql.TrivialDataFetcher;
+import graphql.schema.DataFetcher;
+
+/**
+ * Marker interface to signal that a {@link DgsData} annotated method is "trivial", that is
+ * that the underlying {@link DataFetcher} should implement {@link TrivialDataFetcher}. Useful
+ * to connote simple data fetcher methods that don't do any I/O for instance; such methods are
+ * often handled specially by instrumentation.
+ *
+ * @see TrivialDataFetcher
+ */
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+public @interface Trivial { }
```

Comment:
Have you considered making it a new argument for `@DgsData`? As long as it has a default value, that would be backward compatible as well I think?

Reply:
I recall considering that approach, but I couldn't think of a clean way to make `DgsQuery` work, although I guess I could add it to `DgsQuery` as well and annotate the field with `@AliasFor` to make that work. Having multiple `DgsData` annotations via `@DgsData.List` is also another thing I'd have to make sure works correctly. I can experiment with that and see if it works.

Reply:
@paulbakker okay, I've updated the PR to just use fields on the existing DgsData / DgsQuery annotations, and added test cases for the various scenarios.

Reply:
Very nice! I think I prefer this approach, WDYT?

Reply:
Yeah, I prefer this over adding a new annotation.